restapi/src/main/java/io/stargate/web/resources/v2/RowsResource.java

```diff
@@ -262,6 +264,97 @@ public Response getRows(
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 403, message = "Forbidden", response = Error.class),
+        @ApiResponse(code = 404, message = "Not Found", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal Server Error", response = Error.class)
+      })
+  @Path("/rows")
+  public Response getAllRows(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(value = "Name of the table to use for the request.", required = true)
+          @PathParam("tableName")
+          final String tableName,
+      @QueryParam("fields") final String fields,
+      @ApiParam(value = "Restrict the number of returned items") @QueryParam("page-size")
+          final int pageSizeParam,
+      @ApiParam(value = "Move the cursor to a particular result") @QueryParam("page-state")
+          final String pageStateParam,
+      @ApiParam(value = "Unwrap results", defaultValue = "false") @QueryParam("raw")
+          final boolean raw,
+      @ApiParam(value = "Keys to sort by") @QueryParam("sort") final String sort,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          ByteBuffer pageState = null;
+          if (pageStateParam != null) {
+            byte[] decodedBytes = Base64.getDecoder().decode(pageStateParam);
+            pageState = ByteBuffer.wrap(decodedBytes);
+          }
+
+          int pageSize = DEFAULT_PAGE_SIZE;
+          if (pageSizeParam > 0) {
+            pageSize = pageSizeParam;
+          }
+
+          Map<String, String> allHeaders = getAllHeaders(request);
+          AuthenticatedDB authenticatedDB =
+              db.getDataStoreForToken(token, pageSize, pageState, allHeaders);
+
+          BoundQuery query =
+              authenticatedDB
+                  .getDataStore()
+                  .queryBuilder()
+                  .select()
+                  .from(keyspaceName, tableName)
+                  .build()
+                  .bind();
+
+          final ResultSet r =
+              db.getAuthorizationService()
+                  .authorizedDataRead(
+                      () ->
+                          authenticatedDB
+                              .getDataStore()
+                              .execute(query, ConsistencyLevel.LOCAL_QUORUM)
+                              .get(),
+                      authenticatedDB.getAuthenticationSubject(),
+                      keyspaceName,
+                      tableName,
+                      Collections.emptyList(),
+                      SourceAPI.REST);
```

Comment:
Would it be possible to modify how the where is applied and just reuse `io.stargate.web.resources.v2.RowsResource#getRows(java.lang.String, boolean, java.lang.String, io.stargate.web.resources.AuthenticatedDB, io.stargate.db.schema.Table, java.util.List<io.stargate.db.query.builder.BuiltCondition>)`?

Reply:
Or just refactor that method so it can be used with this as well?